File: searchnsort/ltcd_2331.py
'''
LeetCode - problem 2331
'''

    def evaluateTree(self, root):
        if root.val <= 1: return root.val
        left, right = self.evaluateTree(root.left), self.evaluateTree(root.right)
        return (left or right) if root.val == 2 else (left and right)

File: searchnsort/ltcd_2331.cpp
//LeetCode - problem 2331

    bool evaluateTree(TreeNode* root) {
        if (root->left==NULL) return root->val;
        if (root->val==2) return evaluateTree(root->left)||evaluateTree(root->right);
        return evaluateTree(root->left)&&evaluateTree(root->right);
    }
